### ISSUE TITLE
Unable to change default subnet's "Enable auto-assign public IPv4 address" value

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#4291]
I am unable to change a default subnet's "Enable auto-assign public IPv4 address" value (`map_public_ip_on_launch` attribute).

```hcl
provider "aws" {
  region = "eu-west-2"
}

resource "aws_default_subnet" "d" {
    availability_zone = "eu-west-2a"

    map_public_ip_on_launch = false
}
```
```bash
$ terraform plan
1 error(s) occurred:

* aws_default_subnet.d: "map_public_ip_on_launch": this field cannot be set
```

This value can be changed via the AWS CLI or console.

**Comment by ewbankkit** [comment]
This is because the `aws_default_subnet.map_public_ip_on_launch ` attribute is marked as "output-only": https://github.com/terraform-providers/terraform-provider-aws/blob/4e06a36422cac377c9c4f5fbd573954b857ef38f/aws/resource_aws_default_subnet.go#L38-L42

**Comment by ewbankkit** [comment]
Should also revisit `ipv6_cidr_block` and `assign_ipv6_address_on_creation` attributes once https://github.com/terraform-providers/terraform-provider-aws/issues/4290 has been addressed.

**Comment by bflad** [comment]
Support for this has been merged in via #4396 and will release with v1.17.0 of the AWS provider tomorrow.
